File: src/Billing.Config/TenantConfigSync/Dependencies/TenantSelection.cs
namespace Billing.Config.TenantConfigSync.Dependencies;

/// <summary>One row of tenants.yml: a tenant this instance loads, and its active profile.</summary>
public sealed record SelectedTenant(string Name, bool Enabled, string Profile);

/// <summary>The parsed tenants.yml — which tenants load and their active profiles.</summary>
public sealed class TenantSelection
{
    public IReadOnlyList<SelectedTenant> Tenants { get; init; } = [];

    public IEnumerable<SelectedTenant> Enabled => Tenants.Where(t => t.Enabled);
}

File: java/src/main/java/com/telcobright/billing/tenantconfigsync/dependencies/TenantSelection.java
package com.telcobright.billing.tenantconfigsync.dependencies;

import java.util.List;
import java.util.stream.Collectors;

/**
 * The parsed tenants.yml — which tenants load and their active profiles.
 *
 * <p>Faithful-port note: C# {@code init} property → public field; the computed
 * {@code Enabled} property → the no-arg method {@link #Enabled()}.
 */
public final class TenantSelection {
    public List<SelectedTenant> Tenants = List.of();

    public List<SelectedTenant> Enabled() {
        return Tenants.stream().filter(SelectedTenant::Enabled).collect(Collectors.toList());
    }
}
